apps/hannk/interpreter/model.h

```diff
@@ -92,9 +92,20 @@ class Tensor {
     std::string name_;
     HalideBuffer<void> buffer_;
     QuantizationInfo quantization_;
+    // If true, this Tensor should be considered constant and should not be mutated.
+    // (It may actually refer to read-only external memory, or it may simply be marked this may as
+    // the result of a transform.)
     bool is_constant_ = false;
+    // If true, this Tensor's storage is externally owned and must not be freed.
+    bool is_external_ = false;
```

Comment:
Almost all uses of this are either asserts, or in the delegate, where we could maybe check this from the TFlite side, should we try to do that instead?

If is_external is true, is is_input_ || is_output_ true?

Reply:
> Almost all uses of this are either asserts, or in the delegate, where we could maybe check this from the TFlite side, should we try to do that instead?

I think that using it in asserts is reason enough for it to exist -- the variations in Tensor usage are pretty delicate, and I'd like to keep this validation in place. (I did downgrade most of the HCHECK usage to assert...)

> If is_external is true, is is_input_ || is_output_ true?

For current usage, yes, but there's no intrinsic reason that it has to be.